app/ui.js

```diff
@@ -435,6 +440,16 @@ var UI;
                     document.documentElement.classList.add("noVNC_disconnecting");
                     break;
                 case 'disconnected':
+                    var reconnect = WebUtil.getConfigVar('reconnect');
+
+                    if ((reconnect === 'true' || reconnect == '1') && UI.reconnect_enabled) {
+                        UI.showStatus(_("Reconnecting"));
+
+                        var delay = parseInt(WebUtil.getConfigVar('reconnect_delay', UI.reconnect_delay));
+                        setTimeout(UI.connect, delay);
```

Comment:
For a) a dialogue with the option to cancel would probably be nice, and for b) we could use the transition screen that is used while connecting and disconnecting.